valhalla/baldr/datetime.h

```diff
@@ -31,7 +32,7 @@ struct tz_db_t {
   const date::time_zone* from_index(size_t index) const;
 
 protected:
-  std::vector<std::string> names;
+  std::unordered_map<std::string, size_t> names;
```

Comment:
small refactor to use a hash map to look up the index from the name rather than `std::find`. probably not needed but if we do this alot somewhere it will be way faster than scanning 300+ entries